### PR TITLE
[plugin-web-app] Add new proxy step for using mocks

### PR DESCRIPTION
**Pull request by abudevich**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java [hunk 1]
```diff
@@ -418,14 +438,23 @@ else if (content instanceof byte[])
         throw new IllegalArgumentException("Unsupported content type: " + content.getClass());
     }
 
-    private void applyUrlFilter(StringComparisonRule comparisonRule, String url,
+    private void applyUrlFilter(Set<HttpMethod> httpMethods, StringComparisonRule comparisonRule, String url,
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java:441]: Maybe List<Predicate<MesageInfo>> messageFilters?
mesageFilters.stream().allMatch(p -> p.test(messageInfo))

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java [hunk 2]
```diff
@@ -347,55 +349,82 @@ public void addHeadersToProxyRequest(StringComparisonRule comparisonRule, String
...
+     * In this case no actual request will be executed.
+     * @param httpMethods    The "or"-separated set of HTTP methods to filter by, e.g. 'GET or POST or PUT'
+     * @param comparisonRule String comparison rule: "is equal to", "contains", "does not contain", "matches"
+     * @param url            The string value of URL to match comparison rule
+     * @param responseCode   The code will be used in response
+     * @param content        The content will be used in the response
+     * @param headers        ExamplesTable representing the list of the headers with columns "name" and
+     *                       "value" specifying HTTP header names and values respectively
+     */
+    @When(value = "I mock HTTP responses with HTTP $httpMethods and request URL which $comparisonRule `$url` using"
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java:401]: When I mock HTTP GET or POST responses....?

Comment (abudevich) [reply vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java:401]: > When I mock HTTP GET or POST responses....?

`When I mock HTTP $httpMethods responses with request URL which $comparisonRule `$url` using response code `$responseCode`, content `$payload` and headers:$headers`
?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java [hunk 3]
```diff
@@ -347,55 +350,88 @@ public void addHeadersToProxyRequest(StringComparisonRule comparisonRule, String
...
+     * @param headers        ExamplesTable representing the list of the headers with columns "name" and
+     *                       "value" specifying HTTP header names and values respectively
      */
     @When("I mock HTTP responses with request URL which $comparisonRule `$url` using response code `$responseCode`"
             + " and headers:$headers")
     public void mockHttpRequests(StringComparisonRule comparisonRule, String url, int responseCode,
             DefaultHttpHeaders headers)
     {
-        applyUrlFilter(comparisonRule, url, request -> {
+        applyUrlFilter(List.of(m -> comparisonRule.createMatcher(url).matches(m.getUrl())), request -> {
```
Comment (uarlouski) [inline vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java:382]: compiling regex is not cheap operation

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**